### DISCUSSION TITLE
open_mfdataset not working as expected with rasterio engine?

### DISCUSSION THREAD

**Discussion by pstefa1707** [pydata/xarray#6231] [Q&A]
As the title says, using `open_dataset` with the `engine="rasterio"` kwarg works as expected.

However, using `open_mfdataset` throws a `FileNotFoundError`. Note my raster datasets are on `s3` and are gunzipped.

> ![image](https://user-images.githubusercontent.com/7734121/152094872-7ff30eeb-a5bc-44d0-bcda-2efd6a9cdaf6.png)

Is there something I am doing wrong? thanks for the help!

Edit:
Interesting, if I add the "chunks" parameter to `open_dataset` I will encounter the same `FileNotFoundError`

**Comment by debroize** [comment]
Have the same problem. Did you find a solution?
Best regards

**Reply by pstefa1707** [reply]
Unfortunately cannot remember :/

**Comment by scottyhq** [comment]
Since you're using GDAL paths, for this particular case you can create a [GDAL VRT](https://gdal.org/programs/gdalbuildvrt.html) and open that: 

```python
!gdalbuildvrt -separate stack.vrt {vsipath1} {vsipath2} 
ds = xr.open_dataset('stack.vrt', engine='rasterio')
```

But yes, you're right it is a bit confusing that open_dataset works with a vsipath but open_mfdataset does not. Seems like the logic for figuring out if the string is a local or remote path is slightly different from the full traceback: 

```pytb
File ~/miniforge3/envs/xarray/lib/python3.11/site-packages/xarray/backends/api.py:230, in _get_mtime(filename_or_obj)
    227     path = None
    229 if path and not is_remote_uri(path):
--> 230     mtime = os.path.getmtime(os.path.expanduser(filename_or_obj))
    232 return mtime

File <frozen genericpath>:55, in getmtime(filename)

FileNotFoundError: [Errno 2] No such file or directory:  ...
```
